<launch>

  <!-- Get arguments -->
  <arg name="model_path" default="$(find grounding_sam_ros)weights/groundingdino_swint_ogc.pth" doc="Detection Model Path [default: weights/groundingdino_swint_ogc.pth]" />
  <arg name="config" default="$(find grounding_sam_ros)src/groundingdino/config/GroundingDINO_SwinT_OGC.py" doc="Config path [default: src/groundingdino/config/GroundingDINO_SwinT_OGC.py]" />
  <arg name="box_threshold" default="0.35" doc="Min confidence threshold for box." />
  <arg name="text_threshold" default="0.25" doc="Min confidence threshold for text." />
  <arg name="save" default="False" doc="Save the output images." />
  <arg name="save_path" default="./rgb" doc="Path to save the output images." /> 
  

  <!-- Run Contact-GraspNet -->
  <node name="grounding_sam_ros" pkg="grounding_sam_ros" type="dino_node.py" output="screen" >
      <param name="model_path" type="str" value="$(arg model_path)" />
      <param name="config" type="str" value="$(arg config)" />
      <param name="box_threshold" type="double" value="$(arg box_threshold)" />
      <param name="text_threshold" type="double" value="$(arg text_threshold)" />
      <param name="save" type="bool" value="$(arg save)" />
      <param name="save_path" type="str" value="$(arg save_path)" />
  </node>
</launch>